### PR TITLE
feat(upgrade-dependencies): allow customizing dependency types and semantic commit

### PR DESCRIPTION
**Pull request by iliapolo**
Users can now customize specific dependency types (e.g `dev`, `peer`, ...) as well as the semantic commit type (e.g `chore`, `feat`, ...)

This, along with the [`releasableCommits`](https://github.com/projen/projen/blob/2259df0eb7aea3f79e6aa97ede751c414bbdab97/src/version.ts#L131) feature will help us create separate workflows for runtime / non runtime dependencies and prevent non prod dependencies from triggering a release. 

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/javascript/upgrade-dependencies.ts [hunk 1]
```diff
@@ -24,6 +24,36 @@ import { TaskStep } from "../task-model";
...
+   * `dependencies`
+   */
+  PROD = "prod",
+
+  /**
+   * `bundledDependencies`
+   */
+  BUNDLE = "bundle",
+}
+
```
Comment (mrgrain) [inline src/javascript/upgrade-dependencies.ts:56]: Can use the generic types from there instead and map them later to the NodeJS types
https://github.com/projen/projen/blob/2259df0eb7aea3f79e6aa97ede751c414bbdab97/src/dependencies.ts#L287

**Review by mrgrain — CHANGES_REQUESTED** [review]

File: src/javascript/upgrade-dependencies.ts (hunk 1 above)
Comment (iliapolo) [reply src/javascript/upgrade-dependencies.ts]: Done

File: src/dependencies.ts [hunk 2]
```diff
@@ -322,4 +322,11 @@ export enum DependencyType {
    * Available for Node packages
    */
   OVERRIDE = "override",
+
+  /**
+   * An optional dependency that may or may not be installed by the package manager.
+   *
+   * Available for Node packages
```
Comment (mrgrain) [inline src/dependencies.ts:329]:
```suggestion
   * An optional dependency that may be used at runtime if available, but is not required.
   * It is expected to be installed by the consumer.
```

**Review by mrgrain — APPROVED** [review]

File: src/dependencies.ts (hunk 2 above)
Comment (iliapolo) [reply src/dependencies.ts]: > It is expected to be installed by the consumer.

Not accurate. The package manager will install it - but if the installation fails it doesn't fail the install command

**Pull request merged**